### DISCUSSION TITLE
Redirects: a “*” throws “Error parsing `/fr/blog/*`”

### DISCUSSION THREAD

**Discussion by alexisLefebvre** [vercel/next.js#29745] [Help] [answered]
### What version of Next.js are you using?

11.0.0

### What version of Node.js are you using?

I don't know

### What browser are you using?

Firefox

### What operating system are you using?

Linux Ubuntu 20.04

### How are you deploying your application?

Vercel

### Describe the Bug

I followed this part of the documentation: https://github.com/vercel/next.js/blob/canary/docs/api-reference/next.config.js/redirects.md?plain=1#L59

I wrote this:

```js
      {
        source: '/fr/blog/*',
        destination: 'https://blog.example.com',
        permanent: true,
      },
```

The deployment on Vercel threw an error:

> 13:17:28.215 | Error parsing `/fr/blog/*` https://nextjs.org/docs/messages/invalid-route-source

I rewrote the fist line and it worked:

```js
      {
        source: '/fr/blog/(.*)',
        destination: 'https://blog.example.com',
        permanent: true,
      },
```

### Expected Behavior

The redirect should catch all the requests that start with `/fr/blog/` and redirect the users.

I think the documentation should be updated.

### To Reproduce

I'm not a front developer so I'm not sure it will be enough, but this file `next.config.js` should fail:

```js
module.exports = {
  async redirects() {
    return [
      {
        source: '/fr/blog/(.*)',
        destination: 'https://blog.example.com',
        permanent: true,
      },
    ]
  },
}
```

**Comment by ijjk** [comment, marked as answer]
Hi, If you want to match `/fr/blog` itself along with all other paths the source would need to be `source: '/fr/blog(/.*)?'` note: if using the built-in i18n config as well you need to set `locale: false` to allow matching specific locales.

**Reply by alexisLefebvre** [reply]
Hi, sorry my issue wasn't clear. my request was about this part of the documentation that throw the error I mentioned: https://github.com/vercel/next.js/blob/canary/docs/api-reference/next.config.js/redirects.md?plain=1#L59

**Reply by ijjk** [reply]
If you copy that source directly it doesn't cause the error mentioned, the syntax isn't being followed which is `:someParamName*` the colon is required followed by a param name and then an asterisks if it's a catch-all. This is `path-to-regexp` syntax which can be seen more here https://github.com/pillarjs/path-to-regexp

**Reply by alexisLefebvre** [reply]
Thanks. I understand now, sorry for the inconvenience.
